### PR TITLE
data

### PR DESCRIPTION
**Pull request by jonthegeek**
### Common Mistakes to Catch Fast:
- [x] Did you increment the version number?
- [x] Did you re-build and document the package at the very end to make sure all changes were picked up?
- **GP: Test coverage is very low on this, but I'm not fixing in this PR.**

### Tests:
- [x] Do you have tests for all of the corner cases you can think of?

### Documentation:
- [x] Are all functions documented, including arguments and return values?
- [x] Is there at least one example in the documentation of each exported function?
- **README should really be updated in the next PR, but I focused on the vignette here.**

### Code Quality:
- [x] Are all parameters validated?
- [x] Is the parameter that's most likely to be piped into this function the first parameter?
- [x] Are all internal functions prepended with '.'?
- [x] Did you deal with all hacky things that you meant to address before PRing?

### Housekeeping:
- [ ] Did you review the author checklist (especially things in bold)?

### Documentation:
- [ ] Can you understand what this feature is supposed to do from its help?
- [ ] Does the example seem like a typical use case?

### Code Quality:
- [ ] Can you easily determine what each section of code does?

### Tests:
- [ ] Can you build the package with your personal Rstudio setup?
- [ ] Do the tests pass for you?
- [ ] Can you think of any missing tests?

### CODE REVIEW

**Review by jonathanbratt — CHANGES_REQUESTED** [review]
Some of the comments are just things to think about later, but there are a couple things we might want to change now. Overall, this is *much* cleaner than it was! :)

File: R/vocab.R [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+#'
+#' @return The vocab as a named integer vector. Names are tokens in vocabulary,
+#'   values are integer indices. The casedness of the vocabulary is inferred
+#'   and attached as the "is_cased" attribute.
+#'
+#'   Note that from the perspective of a neural net, the numeric indices *are*
+#'   the tokens, and the mapping from token to index is fixed. If we changed the
+#'   indexing, it would break any pre-trained models using that vocabulary. This
+#'   is why the vocabulary is stored as a named integer vector, and why it
+#'   starts with index zero.
```
[the remark this answers is not in the document]
Comment (jonthegeek) [reply R/vocab.R:60]: See #8 and make sure that captures what you're thinking.

File: vignettes/generating_vocab.Rmd [hunk 2]
```diff
@@ -18,155 +18,154 @@ knitr::opts_chunk$set(
...
 library(wikimorphemes)
+library(wordpiece.data)
 library(dplyr)
 library(ggplot2)
+library(purrr)
 ```
 
 This vignette shows how to use the processed words from the {wikimorphemes}
-package to create a morphemepiece vocabulary and lookup table.
+package to create a {morphemepiece} vocabulary and lookup table.
```
[the remark this answers is not in the document]
Comment (jonthegeek) [reply vignettes/generating_vocab.Rmd:28]: Good call! Updated.

File: R/vocab.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonathanbratt) [reply R/vocab.R:60]: yup! (but switch 1/0 there)
